dev/core#4174 Send/Process Validation errors to FormBuilder front-end

shaneonabike opened the pull request:
Overview
----------------------------------------
As per [dev/core#4174](https://lab.civicrm.org/dev/core/-/issues/4174) in Wordpress the server side validation (from extensions) is not shown on the front-end. The user has the impression things worked when they didn't.

Before
----------------------------------------

User sees that all was well and the form went through

After
----------------------------------------

The user sees a message indicating a failure

Technical Details
----------------------------------------

+ The errors are processed and passed through in the actual `response`
+ The Afform checks for any errors and essentially passes them off to the user
+ Refactored the error handling to be placed into a single `handleError` function

Comments
----------------------------------------

+ I added the code into the Draft processing, but I don't really have time to test it. So if someone wants to give that a test that would be good. I tried to also set the draft back to `unsaved` so that it wasn't in some weird state after an error

shaneonabike committed:
Send/Process Validation errors to FormBuilder front-end

shaneonabike committed:
Reworking Afform validation messages and handling

shaneonabike commented:
@colemanw So I migrated the changes but the issue here is two fold:

+ Validation errors never come back to the front-end and simply a generic message is displayed
+ Standardize the way these messages are appearing
+ I changed the message title a bit because it was a bit scary `Form Error`

shaneonabike committed:
Fixing formatting

colemanw commented:
@shaneonabike this looks like an improvement! I'm still not sure though why there are two calls to `handleError` from the `.then()` callbacks which should not actually fire if there's an error?

shaneonabike commented:
Oh boy~ I went on a bit of an egg hunt on this one. I actually rested everything thinking there was an issue. Check the code again it doesn't have any additional code in the success portion only in the `catch`. Before yes I had added that but in this version I removed it because the catch does work. I just refactored it to 

+ Use `Swal`
+ Allow for proper messaging depending on the type of `error_code` via line 467. Basically, if it's a validation it comes back not as a error `1` so it indicates that it's a validation error which is nicer.

cc @colemanw

shaneonabike committed:
Removing checks in completion
